flink-clients/src/main/java/org/apache/flink/client/deployment/ClusterClientJobClientAdapter.java

```diff
@@ -48,25 +54,45 @@
 
 	private final JobID jobID;
 
-	private final Thread shutdownHook;
+	private final AtomicBoolean running = new AtomicBoolean(true);
 
-	public ClusterClientJobClientAdapter(final ClusterClient<ClusterID> clusterClient, final JobID jobID, final boolean withShutdownHook) {
+	public ClusterClientJobClientAdapter(
+			final ClusterClient<ClusterID> clusterClient,
+			final JobID jobID) {
 		this.jobID = checkNotNull(jobID);
 		this.clusterClient = checkNotNull(clusterClient);
-
-		if (withShutdownHook) {
-			shutdownHook = ShutdownHookUtil.addShutdownHook(
-					clusterClient::shutDownCluster, clusterClient.getClass().getSimpleName(), LOG);
-		} else {
-			shutdownHook = null;
-		}
 	}
 
 	@Override
 	public JobID getJobID() {
 		return jobID;
 	}
 
+	@Override
+	public CompletableFuture<JobStatus> getJobStatus() {
+		return clusterClient.getJobStatus(jobID).thenApply(org.apache.flink.runtime.jobgraph.JobStatus::toCoreJobStatus);
+	}
+
+	@Override
+	public CompletableFuture<Map<String, OptionalFailure<Object>>> getAccumulators(ClassLoader classLoader) {
```

Comment:
Why not use `Optional` here? It seems `OptionalFailure` is a bit strange because it does not behave like an `Optional`, it's more like an `Either` type.

Reply:
Actually, I think it's better to just return a `CompletableFuture<Map<String, Object>>`. We can fail the future if there is any failure in the map but otherwise just return a complete map. I don't think it's useful for the user to have to do all the unpacking. What do you think?

Reply:
We use `Map<String, OptionalFailure<Object>>` as type of accumulators in `JobExecutionResult` & `ClusterClient` already. The investigation how we treat of such unpack things is independent and out of the scope here in my opinion. Shall we open a ticket for take it into consideration and prevent this one to be too complex?